common/path.h

```diff
@@ -105,7 +105,8 @@ absl::Status GetKnownFolderPath(FolderId folder_id, std::string* path);
 // On Windows, variables are matched case invariantly. Unknown environment
 // variables are not changed.
 // On Linux, performs a shell-like expansion. Returns an error if multiple
-// results would be returned, e.g. from *.txt.
+// results would be returned, e.g. from *.txt. Note that the underlying function
+// 'wordexp' allows command execution, e.g. "$(sql DROP TABLE students)".
 absl::Status ExpandPathVariables(std::string* path);
```

Comment:
I don't want to limit ourselves because of potential future code that might misuse it. Sounds like YAGNI to me,  given the current (advanced) state of the project. I don't think it could save us cycles, but rather users of the tool, and we'd likely never find out. I agree, though, that the name is too weak. I'll rename it to ShellExpand, which I think properly describes what the command does on both Windows and Linux, and it hints at the possibility of command substitution.

Reply:
To the contrary: Supporting command expansions without needing it has YAGNI smell to me. This code base already has 69 forks. I would opt for the safer default to not support command expansion. If someone needs it for a reasonable use-case, we can still enable it (maybe with a special flag to make sure that it was the user's intention).